### ISSUE TITLE
VideoTextureView.setMirror should short-circuit if the value is the same

### ISSUE THREAD

**Issue by edenman** [twilio/video-quickstart-android#650]
### Description

Here's the current code:
```
    public void setMirror(final boolean mirror) {
        eglRenderer.setMirror(mirror);
        this.mirror = mirror;
        requestLayout();
    }
```
This always calls eglRenderer and always calls requestLayout, which is pretty wasteful if the new value is the same as the old value.  I can work around this by checking the value before I set it, but that's gross and annoying

### Steps to Reproduce

```
class MyParticipantView {
  fun set(myParticipant: MyParticipant) {
    binding.video.mirror = myParticipant.isLocal
  }
}
```

Easy fix: just add a `if (mirror == this.mirror) return` as the first line of the method.

### Versions

Twilio Video 6.3.0

**Comment by Alton09** [comment]
Hi @edenman . Thanks for reaching out regarding this issue! Good catch. Do you notice a difference in terms of wasted resources in your application when avoiding this call when it's redundant? I'll add this check and will be sure to include in an upcoming release.

**Comment by edenman** [comment]
I've had some issues with video views flickering and this stuck out as a potential culprit.  Not 100% sure it's to blame though, I'm actively developing this feature and it could just be a RecyclerView issue with not reusing `MyParticipantView` views properly.

**Comment by Alton09** [comment]
I see. Maybe take a look at the [twilio video app](https://github.com/twilio/twilio-video-app-android) to get an idea of how to avoid flickering in a video thumbnail list. I recently improved the way the adapter adds video sinks on [this branch](https://github.com/twilio/twilio-video-app-android/tree/feature/media-optimization) and will be merged to master in the next few weeks.

Also, I added the mirror boolean check as you suggested in our active development branch in the Video SDK and all is looking good. Keep an eye out on this issue as I will close it once we have a release out containing this fix. It should be available in a few weeks.
